### PR TITLE
Add Assert template

### LINKED ISSUE #1603: Java: improve Assert
Following this PR by @DamnClin : https://github.com/jhipster/jhipster-lite/pull/1564

The goal here is to 'template' these class, to generate them in a new project.

### PR DESCRIPTION
**Pull request by DamnClin**
Closes #1603 , #1607, #1606

### CODE REVIEW

**Comment by DamnClin** [comment]
Work done with  @sstan001, she will claim the bounties :)

**Comment by pascalgrimaud** [comment]
ok, as soon as it's reviewed and merged, you can @sstan001

File: src/main/resources/generator/server/javatool/base/Assert.java.mustache [hunk 1]
```diff
@@ -1,33 +1,285 @@
 package {{packageName}}.error.domain;
 
+import java.math.BigDecimal;
+import java.time.Instant;
 import java.util.Collection;
+import java.util.Objects;
 
+/**
+ * This class provide utilities for input assertions.
```
Comment (sstan001) [inline src/main/resources/generator/server/javatool/base/Assert.java.mustache:9]: provides

**Review by pascalgrimaud — APPROVED** [review]

**Comment by sstan001** [comment]
I claimed the money 
--> cf https://opencollective.com/generator-jhipster/expenses/76402

**Comment by pascalgrimaud** [comment]
@sstan001 : approved

**Pull request merged**